pkg/build/generator/generator_test.go

```diff
@@ -1732,6 +1732,69 @@ func mockBuildGeneratorForInstantiate() *BuildGenerator {
 			},
 		}, nil
 	}
+	g.Client = c
+	return g
+}
+
+func mockBuildGeneratorForDockerNoCache() *BuildGenerator {
```

Comment:
this is a pretty ugly pattern, how about making a mockBuildGenerator() that takes all the potential func overrides as args (and if you pass nil, does not override that func)?

otherwise we're just going to have an ever growing set of mockBuildGeneratorForXXX funcs w/ a bunch of cut+paste.

Reply:
Refactored mockBuildGenerator per your suggestion, ptal